pkg/dao/migration/trigger_modelchange.go

```diff
@@ -37,40 +37,59 @@ CREATE OR REPLACE FUNCTION {{ .Function }}() RETURNS TRIGGER
 AS 
 $$
 DECLARE
-    ids        json;
-    ids_length integer;
-    payload    text;
+    ids                 jsonb;
+    ids_length          int;
+    payload             text;
+    has_environment_id  boolean;
+    has_project_id      boolean;
 BEGIN
-    
-    -- Select ID list from transition table,
-    -- and convert the result to a JSON string array.
+    ids := '[]'::jsonb;
+    has_project_id := false;
+    has_environment_id := false;
+
+    IF EXISTS (
+        SELECT 1 FROM information_schema.columns WHERE table_name = TG_TABLE_NAME AND column_name = 'project_id'
+    ) THEN
+        has_project_id := true;
+    END IF;
+
+    IF EXISTS (
+        SELECT 1 FROM information_schema.columns WHERE table_name = TG_TABLE_NAME AND column_name = 'environment_id'
+    ) THEN
+        has_environment_id := true;
+    END IF;
```

Comment:
select count(1) from ... where .... and column_name in ('project_id', 'environment_id')

-- 0: nothing
-- 1: project_id
-- 2: both

Reply:
updated